src/components/EmailNotifications/EmailNotificationsProvider.js

```diff
@@ -0,0 +1,142 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
+import PropTypes from 'prop-types'
+import { useWallet } from '../../providers/Wallet'
+import {
+  getJurorEmail,
+  getSubscriptionDetails,
+  subscribeExistingEmail,
+  subscribeToNotifications,
+} from '../../services/servicesRequests'
+
+const EmailNotificationsContext = React.createContext()
+
+function EmailNotificationsProvider({ children }) {
+  const { account } = useWallet()
+
+  const asyncCancelled = useRef(false)
+  const [email, setEmail] = useState(null)
+  const [needsSignature, setNeedsSignature] = useState(false)
+  const [subscriptionDetails, setSubscriptionDetails] = useState({
+    error: null,
+    fetching: false,
+  })
+
+  const handleOnSubscribe = useCallback(
+    async email => {
+      const response = await subscribeToNotifications(account, email)
+
+      if (response.error && !response.needsSignature) {
+        return response.error
+      }
+
+      if (!asyncCancelled.current) {
+        setEmail(email)
+        if (needsSignature !== Boolean(response.needsSignature)) {
+          setNeedsSignature(!needsSignature)
```

Comment:
Trying to figure out how are you thinking to handle the action after the signature? i mean after the signature is successful we need to call again the `subscribeToNotifications` again.
And the other question that i have is : given that here we don't have an useEffect that listen to the `needsSignature` state,  i guess that the plan is to have that useEffect in the manager?

Reply:
> given that here we don't have an useEffect that listen to the needsSignature state, i guess that the plan is to have that useEffect in the manager?

Yes, exactly, that was something I wanted to leave to the state machine manager, given that we could have different machines based on the area (preferences or modal).

> Trying to figure out how are you thinking to handle the action after the signature?

Similar to above, I was hoping the manager would handle this. These functions just attempt to "optimistically" call the API, and if it fails, it'll let us know either with the error or a state change.